component/attr_cache/cacheMap_test.go

```diff
@@ -0,0 +1,377 @@
+   |     |  |      |     | |     | |     |     | |       |            |
+   |     |  |      |     | |     | |     |     | |       |            |
+   | --- |  |      |     | |-----| |---- |     | |-----| |-----  ------
+   |     |  |      |     | |     | |     |     |       | |       |
+   | ____|  |_____ | ____| | ____| |     |_____|  _____| |_____  |_____
+
+
+   Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+
+   Copyright © 2023 Seagate Technology LLC and/or its Affiliates
+   Copyright © 2020-2023 Microsoft Corporation. All rights reserved.
+   Author : <blobfusedev@microsoft.com>
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE
+*/
+
+package attr_cache
+
+import (
+	"container/list"
+	"testing"
+	"time"
+
+	"github.com/Seagate/cloudfuse/internal"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+// look at s3wrappers tests for reference. look at common folder tests as well for reference. half way between s3wrappers and utils test
+
+type cacheMapTestSuite struct {
+	suite.Suite
+	assert            *assert.Assertions
+	rootAttrCacheItem attrCacheItem
+}
+
+// what is every test going to need to test with?
+func (suite *cacheMapTestSuite) SetupTest() {
+	suite.assert = assert.New(suite.T())
+
+	rootAttr := internal.CreateObjAttrDir("")
+	suite.rootAttrCacheItem = *newAttrCacheItem(rootAttr, true, time.Now())
+
+	//set up nested Dir tree
+	nestedDir, nestedFiles := generateNestedDirectory("a")
+
+	for dir := nestedDir.Front(); dir != nil; dir = dir.Next() {
+		attr := internal.CreateObjAttrDir(dir.Value.(string))
+		suite.rootAttrCacheItem.insert(attr, true, time.Now())
+	}
+
+	for file := nestedFiles.Front(); file != nil; file = file.Next() {
+		attr := internal.CreateObjAttr(file.Value.(string), 1024, time.Now())
+		suite.rootAttrCacheItem.insert(attr, true, time.Now())
+	}
+
+}
+
+func (suite *cacheMapTestSuite) TestInsertFileCacheMap() {
```

Comment:
```suggestion
func (suite *cacheMapTestSuite) TestInsertFile() {
```
No need to name the struct we're testing.
This needs to be fixed throughout the tests.